src/state/erc20/hooks.ts

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useMemo } from 'react'
+
+import { TokenErc20 } from '@gnosis.pm/dex-js'
+
+import { Network } from 'types'
+
+import useGlobalState from 'hooks/useGlobalState'
+
+import { buildErc20Key, Erc20State } from '.'
+import { saveMultipleErc20 } from './actions'
+
+export type SingleErc20State = TokenErc20 | null
+
+type UseErc20Params = {
+  address?: string
+  networkId?: Network
+}
+
+/**
+ * Syntactic sugar to get erc20 from global state
+ */
+export const useErc20 = <State extends { erc20s: Erc20State }>(params: UseErc20Params): SingleErc20State => {
+  const { address, networkId } = params
+
+  const [{ erc20s }] = useGlobalState<State>()
+
+  return useMemo(() => {
+    if (!address || !networkId) {
+      return null
+    }
+    return erc20s.get(buildErc20Key(networkId, address)) || null
+  }, [address, erc20s, networkId])
+}
+
+type UseMultipleErc20Params = {
+  addresses: string[]
+  networkId?: Network
+}
+
+/**
+ * Syntactic sugar to get erc20s from global state
+ */
+export const useMultipleErc20s = <State extends { erc20s: Erc20State }>(
+  params: UseMultipleErc20Params,
+): Record<string, SingleErc20State> => {
+  const { addresses, networkId } = params
+  const [{ erc20s }] = useGlobalState<State>()
+
+  return useMemo(() => {
+    if (!networkId) {
+      // Return map of address => null
+      return addresses.reduce((acc, address) => ({ ...acc, [address]: null }), {})
+    }
+
+    return addresses.reduce(
+      (acc, address) => ({ ...acc, [address]: erc20s.get(buildErc20Key(networkId, address)) || null }),
```

Comment:
a bit hard to read, but not that bad. I personally find is ok more line breaks...